webapp/src/validation/preValidation.ts

```diff
@@ -1,9 +1,40 @@
-export function preValidate(value: any): string[] { 
-  const errorMessages: string[] = []
+import { ValidationError } from "./validationError";
+
+export function preValidate(row: number, columnName: string, value: any): Set<ValidationError> { 
+  const validationErrors = new Set<ValidationError>()
 
   if (parseInt(value, 10) === 1337) {
-    errorMessages.push("Oops! That value is too cool to be true!");
+    validationErrors.add({
+      index: row,
+      column: columnName,
+      errorMessage: "Oops! That value is too *sick* to be true!"
+    })
+  }
+
+  if (columnName === "DBH"){
+    if (!value) {
+      validationErrors.add({
+        index: row,
+        column: columnName,
+        errorMessage:"Gotta enter a value"
+      })
+    }
+
+    if (parseInt(value, 10) > 200) {
+      validationErrors.add({
+        index: row,
+        column: columnName,
+        errorMessage:"Oops! That value is too high."
```

Comment:
These messages did make me lol, but since this is going to be demoed to the director of ForestGEO, I would appreciate something more professional. I.e. "DBH value must be between 1 and 200."